README.pod
=pod

=head1 NAME

Task::BeLike::DAGOLDEN - DAGOLDEN's frequently-used or favorite modules

=head1 VERSION

version 0.008

=head1 TASK CONTENTS

=head2 Apps

=head3 L<App::Ack>

=head3 L<App::ForkProve>

=head3 L<App::grindperl>

=head3 L<App::mymeta_requires>

=head3 L<App::Nopaste>

=head3 L<App::Uni>

=head3 L<Code::TidyAll>

=head2 CLI tools

=head3 L<Daemon::Daemonize>

=head3 L<Getopt::Lucid>

=head3 L<Pod::Usage>

=head2 Filesystem tools

=head3 L<File::Find::Rule>

=head3 L<File::Find::Rule::Perl>

=head3 L<File::Slurp>

=head3 L<File::pushd>

=head3 L<Path::Class>

=head3 L<Path::Class::Rule>

=head3 L<Path::Iterator::Rule>

=head3 L<Path::Tiny>

=head3 L<Unicode::UTF8>

=head2 Database Modules

=head3 L<DBI>

=head3 L<DBD::SQLite>

=head2 Testing Modules

=head3 L<Test::Deep>

=head3 L<Test::Differences>

=head3 L<Test::Fatal>

Better than Test::Exception

=head3 L<Test::More> 0.96

Version 0.96 required because: Minimally sane subtest()'s

=head3 L<Test::Routine>

=head3 L<Test::Roo>

=head3 L<Test::Perl::Critic>

=head3 L<Perl::Critic::Lax>

=head3 L<Test::CPAN::Meta>

=head3 L<Test::Pod::Coverage> 1.08

=head3 L<Pod::Coverage::TrustPod>

=head3 L<Test::Pod> 1.41

=head3 L<Test::FailWarnings> 0.003

=head3 L<TAP::Harness::Restricted>

=head2 Perl development tools

=head3 L<Acme::require::case>

=head3 L<Archive::Tar>

=head3 L<Archive::Tar::Wrapper>

=head3 L<Archive::Zip>

=head3 L<Benchmark::Forking>

=head3 L<CPAN::DistnameInfo>

=head3 L<CPAN::Mini>

=head3 L<CPAN::Uploader>

=head3 L<CPAN::Visitor>

=head3 L<Data::Dump::Streamer>

=head3 L<Devel::Cover>

=head3 L<Devel::NYTProf>

=head3 L<Dumbbench>

=head3 L<Dist::Zilla> 4

=head3 L<Dist::Zilla::PluginBundle::DAGOLDEN> 0.043

=head3 L<Dist::Zooky>

=head3 L<Module::Load::Conditional>

=head3 L<Perl::Version>

=head3 L<Pod::Strip>

=head3 L<autodie>

=head3 L<namespace::autoclean>

=head3 L<superclass>

=head3 L<version>

=head2 Moose

=head3 L<Moose>

=head3 L<MooseX::Types>

=head3 L<MooseX::Types::Path::Tiny>

=head2 Email tools

=head3 L<Email::MIME>

=head3 L<Email::Sender>

=head3 L<Email::Sender::Simple>

=head3 L<Email::Simple>

=head3 L<Email::Simple::Creator>

=head2 Web tools

=head3 L<Bundle::LWP>

=head3 L<LWP::Protocol::https>

=head3 L<HTTP::Tiny> 0.028

=head3 L<HTTP::CookieJar>

=head3 L<Mozilla::CA>

=head3 L<IO::Socket::SSL>

=head3 L<Plack>

=head3 L<URI>

=head3 L<WWW::Mechanize>

=head2 Data manipulation tools

=head3 L<Const::Fast>

=head3 L<Data::Stream::Bulk>

=head3 L<JSON>

=head3 L<JSON::XS>

=head3 L<List::AllUtils>

=head3 L<Regexp::Common>

=head3 L<Syntax::Keyword::Junction>

=head3 L<XML::RSS>

=head3 L<XML::Simple>

=head3 L<YAML>

=head2 System interaction tools

=head3 L<autodie>

=head3 L<Capture::Tiny>

=head3 L<IO::CaptureOutput>

=head3 L<IO::Prompt::Tiny>

=head3 L<IPC::Run3>

=head3 L<IPC::System::Simple>

=head3 L<Time::HiRes>

=head2 Special-purpose tools

=head3 L<Git::Wrapper>

=head3 L<Image::ExifTool>

=head3 L<Net::GitHub>

=head3 L<Vi::QuickFix>

=head1 AUTHOR

David Golden <dagolden@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is Copyright (c) 2011 by David Golden.

This is free software, licensed under:

  The Apache License, Version 2.0, January 2004



Changes
Revision history for Task-BeLike-DAGOLDEN

{{$NEXT}}

1.011     2018-04-21 22:41:52+02:00 Europe/Oslo

    - removed File::Slurp

1.010     2015-02-02 11:29:16-05:00 America/New_York

    - add cpanlistoutdated, cpanoutdated, Moo, namespace::clean,
      JSON::MaybeXS, YAML::XS

    - removed JSON and JSON::XS

    - bumped version numbers on some dzil modules

    - modernize repo layout

0.009     2014-01-12 09:57:06-05:00 America/New_York

    - remove Data::Dump::Streamer -- it is not well supported and
      keeps failing tests

0.008     2013-08-20 18:08:13 America/New_York

    - add IO::Prompt::Tiny

0.007     2013-06-11 14:41:52 America/New_York

    - require HTTP::Tiny 0.028 for good SSL/cookie-jar support

    - add Dist::Zooky for converting old dists to dzil

    - remove App::cpanminus to avoid interfering with perlbrew one

0.006     2013-03-13 22:44:47 America/New_York

    - superclass, TAP::Harness::Restricted

    - remove Pantry

0.005     2013-03-13 05:56:12 America/New_York

    - Unicode::UTF8, Test::Roo, Test::FailWarnings, HTTP::CookieJar,
      Mozilla::CA, autodie, Net::GitHub, Acme::require::case,
      Benchmark::Forking and Dumbbench

0.004     2013-01-31 11:25:14 America/New_York

    - create a Moose section

    - add Path::Tiny

0.003     2013-01-24 17:25:49 America/New_York
    - a few more modules

0.002     2012-03-30 13:37:22 Europe/Paris
    - many more modules

0.001     2011-10-21 13:22:00 America/New_York
    - Initial release



lib/Task/BeLike/DAGOLDEN.pm
use 5.006;
use strict;
use warnings;

package Task::BeLike::DAGOLDEN;
# ABSTRACT: DAGOLDEN's frequently-used or favorite modules

our $VERSION = '1.012';


=pkgroup Apps

=pkg App::Ack

=pkg App::ForkProve

=pkg App::cpanlistchanges

=pkg App::cpanoutdated

=pkg App::grindperl

=pkg App::mymeta_requires

=pkg App::Nopaste

=pkg App::Uni

=pkg Code::TidyAll


=pkgroup CLI tools

=pkg Daemon::Daemonize

=pkg Getopt::Lucid

=pkg Pod::Usage


=pkgroup Filesystem tools

=pkg File::Find::Rule

=pkg File::Find::Rule::Perl

=pkg File::pushd

=pkg Path::Class

=pkg Path::Class::Rule

=pkg Path::Iterator::Rule

=pkg Path::Tiny

=pkg Unicode::UTF8


=pkgroup Database Modules

=pkg DBI

=pkg DBD::SQLite


=pkgroup Testing Modules

=pkg Test::Deep

=pkg Test::Differences

=pkg Test::Fatal

Better than Test::Exception

=pkg Test::More  0.96   Minimally sane subtest()'s

=pkg Test::Routine

=pkg Test::Roo

=pkg Test::Perl::Critic

=pkg Perl::Critic::Lax

=pkg Test::CPAN::Meta

=pkg Test::Pod::Coverage 1.08

=pkg Pod::Coverage::TrustPod

=pkg Test::Pod 1.41

=pkg Test::FailWarnings 0.003

=pkg TAP::Harness::Restricted


=pkgroup Perl development tools

=pkg Acme::require::case

=pkg Archive::Tar

=pkg Archive::Tar::Wrapper

=pkg Archive::Zip

=pkg Benchmark::Forking

=pkg CPAN::DistnameInfo

=pkg CPAN::Mini

=pkg CPAN::Uploader

=pkg CPAN::Visitor

=pkg Devel::Cover

=pkg Devel::NYTProf

=pkg Dumbbench

=pkg Dist::Zilla 5

=pkg Dist::Zilla::PluginBundle::DAGOLDEN 0.072

=pkg Dist::Zooky

=pkg Module::Load::Conditional

=pkg Perl::Version

=pkg Pod::Strip

=pkg autodie

=pkg namespace::autoclean

=pkg namespace::clean

=pkg superclass

=pkg version


=pkgroup Moose

=pkg Moose

=pkg MooseX::Types

=pkg MooseX::Types::Path::Tiny

=pkg Moo


=pkgroup Email tools

=pkg Email::MIME

=pkg Email::Sender

=pkg Email::Sender::Simple

=pkg Email::Simple

=pkg Email::Simple::Creator


=pkgroup Web tools

=pkg Bundle::LWP

=pkg LWP::Protocol::https

=pkg HTTP::Tiny 0.028

=pkg HTTP::CookieJar

=pkg Mozilla::CA

=pkg IO::Socket::SSL

=pkg Plack

=pkg URI

=pkg WWW::Mechanize


=pkgroup Data manipulation tools

=pkg Const::Fast

=pkg Data::Stream::Bulk

=pkg JSON::MaybeXS

=pkg List::AllUtils

=pkg Regexp::Common

=pkg Syntax::Keyword::Junction

=pkg XML::RSS

=pkg XML::Simple

=pkg YAML

=pkg YAML::XS


=pkgroup System interaction tools

=pkg autodie

=pkg Capture::Tiny

=pkg IO::CaptureOutput

=pkg IO::Prompt::Tiny

=pkg IPC::Run3

=pkg IPC::System::Simple

=pkg Time::HiRes


=pkgroup Special-purpose tools

=pkg Git::Wrapper

=pkg Image::ExifTool

=pkg Net::GitHub

=pkg Vi::QuickFix

=cut

1;

# vim: ts=2 sts=2 sw=2 et:


Makefile.PL
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.17;

my %WriteMakefileArgs = (
  "ABSTRACT" => "DAGOLDEN's frequently-used or favorite modules",
  "AUTHOR" => "David Golden <dagolden\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.17"
  },
  "DISTNAME" => "Task-BeLike-DAGOLDEN",
  "LICENSE" => "apache",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Task::BeLike::DAGOLDEN",
  "PREREQ_PM" => {
    "Acme::require::case" => 0,
    "App::Ack" => 0,
    "App::ForkProve" => 0,
    "App::Nopaste" => 0,
    "App::Uni" => 0,
    "App::cpanlistchanges" => 0,
    "App::cpanoutdated" => 0,
    "App::grindperl" => 0,
    "App::mymeta_requires" => 0,
    "Archive::Tar" => 0,
    "Archive::Tar::Wrapper" => 0,
    "Archive::Zip" => 0,
    "Benchmark::Forking" => 0,
    "Bundle::LWP" => 0,
    "CPAN::DistnameInfo" => 0,
    "CPAN::Mini" => 0,
    "CPAN::Uploader" => 0,
    "CPAN::Visitor" => 0,
    "Capture::Tiny" => 0,
    "Code::TidyAll" => 0,
    "Const::Fast" => 0,
    "DBD::SQLite" => 0,
    "DBI" => 0,
    "Daemon::Daemonize" => 0,
    "Data::Stream::Bulk" => 0,
    "Devel::Cover" => 0,
    "Devel::NYTProf" => 0,
    "Dist::Zilla" => 5,
    "Dist::Zilla::PluginBundle::DAGOLDEN" => "0.072",
    "Dist::Zooky" => 0,
    "Dumbbench" => 0,
    "Email::MIME" => 0,
    "Email::Sender" => 0,
    "Email::Sender::Simple" => 0,
    "Email::Simple" => 0,
    "Email::Simple::Creator" => 0,
    "File::Find::Rule" => 0,
    "File::Find::Rule::Perl" => 0,
    "File::pushd" => 0,
    "Getopt::Lucid" => 0,
    "Git::Wrapper" => 0,
    "HTTP::CookieJar" => 0,
    "HTTP::Tiny" => "0.028",
    "IO::CaptureOutput" => 0,
    "IO::Prompt::Tiny" => 0,
    "IO::Socket::SSL" => 0,
    "IPC::Run3" => 0,
    "IPC::System::Simple" => 0,
    "Image::ExifTool" => 0,
    "JSON::MaybeXS" => 0,
    "LWP::Protocol::https" => 0,
    "List::AllUtils" => 0,
    "Module::Load::Conditional" => 0,
    "Moo" => 0,
    "Moose" => 0,
    "MooseX::Types" => 0,
    "MooseX::Types::Path::Tiny" => 0,
    "Mozilla::CA" => 0,
    "Net::GitHub" => 0,
    "Path::Class" => 0,
    "Path::Class::Rule" => 0,
    "Path::Iterator::Rule" => 0,
    "Path::Tiny" => 0,
    "Perl::Critic::Lax" => 0,
    "Perl::Version" => 0,
    "Plack" => 0,
    "Pod::Coverage::TrustPod" => 0,
    "Pod::Strip" => 0,
    "Pod::Usage" => 0,
    "Regexp::Common" => 0,
    "Syntax::Keyword::Junction" => 0,
    "TAP::Harness::Restricted" => 0,
    "Test::CPAN::Meta" => 0,
    "Test::Deep" => 0,
    "Test::Differences" => 0,
    "Test::FailWarnings" => "0.003",
    "Test::Fatal" => 0,
    "Test::More" => "0.96",
    "Test::Perl::Critic" => 0,
    "Test::Pod" => "1.41",
    "Test::Pod::Coverage" => "1.08",
    "Test::Roo" => 0,
    "Test::Routine" => 0,
    "Time::HiRes" => 0,
    "URI" => 0,
    "Unicode::UTF8" => 0,
    "Vi::QuickFix" => 0,
    "WWW::Mechanize" => 0,
    "XML::RSS" => 0,
    "XML::Simple" => 0,
    "YAML" => 0,
    "YAML::XS" => 0,
    "autodie" => 0,
    "namespace::autoclean" => 0,
    "namespace::clean" => 0,
    "strict" => 0,
    "superclass" => 0,
    "version" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "1.011",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Acme::require::case" => 0,
  "App::Ack" => 0,
  "App::ForkProve" => 0,
  "App::Nopaste" => 0,
  "App::Uni" => 0,
  "App::cpanlistchanges" => 0,
  "App::cpanoutdated" => 0,
  "App::grindperl" => 0,
  "App::mymeta_requires" => 0,
  "Archive::Tar" => 0,
  "Archive::Tar::Wrapper" => 0,
  "Archive::Zip" => 0,
  "Benchmark::Forking" => 0,
  "Bundle::LWP" => 0,
  "CPAN::DistnameInfo" => 0,
  "CPAN::Mini" => 0,
  "CPAN::Uploader" => 0,
  "CPAN::Visitor" => 0,
  "Capture::Tiny" => 0,
  "Code::TidyAll" => 0,
  "Const::Fast" => 0,
  "DBD::SQLite" => 0,
  "DBI" => 0,
  "Daemon::Daemonize" => 0,
  "Data::Stream::Bulk" => 0,
  "Devel::Cover" => 0,
  "Devel::NYTProf" => 0,
  "Dist::Zilla" => 5,
  "Dist::Zilla::PluginBundle::DAGOLDEN" => "0.072",
  "Dist::Zooky" => 0,
  "Dumbbench" => 0,
  "Email::MIME" => 0,
  "Email::Sender" => 0,
  "Email::Sender::Simple" => 0,
  "Email::Simple" => 0,
  "Email::Simple::Creator" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Find::Rule" => 0,
  "File::Find::Rule::Perl" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "Getopt::Lucid" => 0,
  "Git::Wrapper" => 0,
  "HTTP::CookieJar" => 0,
  "HTTP::Tiny" => "0.028",
  "IO::CaptureOutput" => 0,
  "IO::Prompt::Tiny" => 0,
  "IO::Socket::SSL" => 0,
  "IPC::Run3" => 0,
  "IPC::System::Simple" => 0,
  "Image::ExifTool" => 0,
  "JSON::MaybeXS" => 0,
  "LWP::Protocol::https" => 0,
  "List::AllUtils" => 0,
  "Module::Load::Conditional" => 0,
  "Moo" => 0,
  "Moose" => 0,
  "MooseX::Types" => 0,
  "MooseX::Types::Path::Tiny" => 0,
  "Mozilla::CA" => 0,
  "Net::GitHub" => 0,
  "Path::Class" => 0,
  "Path::Class::Rule" => 0,
  "Path::Iterator::Rule" => 0,
  "Path::Tiny" => 0,
  "Perl::Critic::Lax" => 0,
  "Perl::Version" => 0,
  "Plack" => 0,
  "Pod::Coverage::TrustPod" => 0,
  "Pod::Strip" => 0,
  "Pod::Usage" => 0,
  "Regexp::Common" => 0,
  "Syntax::Keyword::Junction" => 0,
  "TAP::Harness::Restricted" => 0,
  "Test::CPAN::Meta" => 0,
  "Test::Deep" => 0,
  "Test::Differences" => 0,
  "Test::FailWarnings" => "0.003",
  "Test::Fatal" => 0,
  "Test::More" => "0.96",
  "Test::Perl::Critic" => 0,
  "Test::Pod" => "1.41",
  "Test::Pod::Coverage" => "1.08",
  "Test::Roo" => 0,
  "Test::Routine" => 0,
  "Time::HiRes" => 0,
  "URI" => 0,
  "Unicode::UTF8" => 0,
  "Vi::QuickFix" => 0,
  "WWW::Mechanize" => 0,
  "XML::RSS" => 0,
  "XML::Simple" => 0,
  "YAML" => 0,
  "YAML::XS" => 0,
  "autodie" => 0,
  "namespace::autoclean" => 0,
  "namespace::clean" => 0,
  "strict" => 0,
  "superclass" => 0,
  "version" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
